cmd/playground/tasks.go

```diff
@@ -228,3 +254,75 @@ func formatTaskStatus(status api.PlayTaskStatus) string {
 func taskIsFinished(task api.PlayTask) bool {
 	return task.Status == api.PlayTaskStatusCompleted || task.Status == api.PlayTaskStatusFailed
 }
+
+func filterTasksByKind(tasks map[string]api.PlayTask, kind string) map[string]api.PlayTask {
+	if kind == "" {
+		return tasks
+	}
+
+	filtered := make(map[string]api.PlayTask)
+	for name, task := range tasks {
+		switch kind {
+		case "init":
+			if task.Init {
+				filtered[name] = task
+			}
+		case "helper":
+			if task.Helper {
+				filtered[name] = task
+			}
+		case "regular":
+			if !task.Init && !task.Helper {
+				filtered[name] = task
+			}
+		}
+	}
+
+	return filtered
+}
+
+func countTasksByKind(tasks map[string]api.PlayTask, kind string) int {
+	if kind == "" {
+		// When no kind is specified, only count non-helper tasks for waiting
+		count := 0
+		for _, task := range tasks {
+			if !task.Helper {
+				count++
+			}
+		}
+		return count
+	}
+
+	filtered := filterTasksByKind(tasks, kind)
+	// For helper tasks, return the full count since we don't wait on them
+	if kind == "helper" {
+		return len(filtered)
+	}
+	// For init and regular, only count those that we actually wait on
+	return len(filtered)
```

Comment:
Is it a copy-paste issue? The return expression is identical to the one in the `if kind == "helper"` block.